mailbox/api/src/main/java/org/apache/james/mailbox/MailboxManager.java

```diff
@@ -153,7 +155,7 @@
      * @throws MailboxException
      *             when creation fails
      */
-    void createMailbox(MailboxPath mailboxPath, MailboxSession mailboxSession) throws MailboxException;
+    Optional<MailboxId> createMailbox(MailboxPath mailboxPath, MailboxSession mailboxSession) throws MailboxException;
```

Comment:
documentation doesn't explain why optional ?
And what about "intermediary mailboxes created" ?